sdk/core/System.ClientModel/src/Options/MessageClassifier.cs

```diff
@@ -22,5 +24,56 @@ public virtual bool IsError(PipelineMessage message)
             int statusKind = message.Response.Status / 100;
             return statusKind == 4 || statusKind == 5;
         }
+
+        // TODO: If we use retry classification as part of the same class as error
+        // classification, we need to do something special to not override user-set
+        // retry classifier when we set error classifier in DPG.  I think we *do*
+        // have to solve this.  Consider a separate class "RetryClassifier" that could
+        // be set in Message property bag if we don't want it in the public API.
```

Comment:
This is a challenge.  We could either look at separating the classifiers for retries and errors into two separate types, or we could say in ClientModel, you have to override the retry policy to update the list of status codes that are retriable and the exceptions that indicate a request should be retried.